src/components/renderer/SvgRenderer.tsx

```diff
@@ -0,0 +1,182 @@
+  }, [document]);
+
+  const setSvgDataAndTriggerCallback = (svgToSet: string) => {
+    setFetchedSvgData(svgToSet);
+    setToDisplay(DisplayResult.OK);
+    setTimeout(() => {
+      updateIframeHeight();
+      if (typeof onConnected === "function") {
+        onConnected();
+      }
+    }, 200); // wait for 200ms before manually updating the height
+  };
+
+  // Step 2: Recompute and compare the digestMultibase if present, if not proceed to use the svg template
+  useEffect(() => {
+    if (!buffer) return;
+
+    const digestMultibaseInDoc = docAsAny.renderMethod.digestMultibase;
+    const svgUint8Array = new Uint8Array(buffer ?? []);
+    const decoder = new TextDecoder();
+    const text = decoder.decode(svgUint8Array);
+
+    if (digestMultibaseInDoc) {
+      const shaDigest = crypto.createHash("sha256").update(svgUint8Array).digest();
+      const recomputedDigestMultibase = "z" + bs58.encode(shaDigest); // manually prefix with 'z' as per https://w3c-ccg.github.io/multibase/#mh-registry
+      if (recomputedDigestMultibase === digestMultibaseInDoc) {
+        setSvgDataAndTriggerCallback(text);
+      } else {
+        setToDisplay(DisplayResult.DIGEST_ERROR);
+      }
+    } else {
+      setSvgDataAndTriggerCallback(text);
+    }
+    /* eslint-disable-next-line react-hooks/exhaustive-deps */
+  }, [buffer]);
+
+  // Step 3: Compile final svg
+  const renderTemplate = (template: string, document: any) => {
+    if (template.length === 0) return "";
+    if (forceV2 && utils.isRawV2Document(document)) {
+      const v2doc = document as v2.OpenAttestationDocument;
+      const compiledTemplate = handlebars.compile(template);
+      return compiledTemplate(v2doc);
+    } else {
+      const v4doc = document;
+      const compiledTemplate = handlebars.compile(template);
+      return compiledTemplate(v4doc.credentialSubject);
+    }
+  };
+
+  const compiledSvgData = `data:image/svg+xml,${encodeURIComponent(renderTemplate(svgFetchedData, document))}`;
+
+  const updateIframeHeight = () => {
+    if (svgRef.current) {
+      const contentHeight = svgRef.current?.contentDocument?.body?.offsetHeight;
+      svgRef.current.style.height = `${contentHeight}px`;
+    }
+  };
+
+  const iframeContent = `
+      <html>
+          <head></head>
+          <body style="margin: 0; display: flex; justify-content: center; align-items: center;">
+          ${renderToStaticMarkup(
+            <>{svgFetchedData ? <img src={compiledSvgData} alt="SVG document image" /> : <></>}</>
+          )}
+          </body>
+      </html>`;
+
+  switch (toDisplay) {
+    case DisplayResult.DEFAULT:
+      return <NoTemplate document={docAsAny} handleObfuscation={() => null} />;
+    case DisplayResult.CONNECTION_ERROR:
+      return <ConnectionFailureTemplate document={document} source={source} />;
+    case DisplayResult.DIGEST_ERROR:
+      return <TamperedSvgTemplate document={document} />;
+    case DisplayResult.OK:
+      return (
+        <iframe
```

Comment:
actually do we really need an iframe, would it be simpler to just return
```
<img src={compiledSvgData} alt="SVG document image" />
```
which is already considered safe to XSS, see https://digi.ninja/blog/svg_xss.php 

if can use the `img` directly without the `iframe`, we could:
1. easily handle malform svgs and render the default renderer instead via `onError` prop on `img`
2. users have more control over the styling of the svg image

Reply:
ok updated